apps/aetestframework/lib/worker.ex

```diff
@@ -1,599 +1,161 @@
+        new_state = Map.put(state, node_name, new_node)
 
-  def handle_call({:send_command, node_name, cmd}, _, state) do
-    if Map.has_key?(state, node_name) do
-      port = state[node_name].process_port
-      Port.command(port, cmd)
-      receive_result(state)
-    else
-      {:reply, :unknown_node, state}
+        {:reply, :ok, new_state}
     end
   end
 
-  def handle_call({:get_balance, node}, _, state) do
-    {:reply, state[node].miner_balance, state}
-  end
-
-  def handle_call({:update_pubkeys_state}, _, state) do
-    new_state = Enum.reduce(state, state, fn(node, acc) ->
-      {node_name, _} = node
-      port = state[node_name].port
-      {:ok, peer_info} = Client.get_info("localhost:#{port}")
-      pubkey = peer_info.public_key
-      put_in(acc[node_name].miner_pubkey, Account.base58c_decode(pubkey))
-    end)
-
-    {:reply, :ok, new_state}
+  def handle_call({:get, node, match_by, cmd}, _from, state) do
+    %{port_id: port_id} = Map.get(state, node)
+    command = "{:#{match_by}," <> cmd <> "}\n"
+    Port.command(port_id, command)
+    result = receive_result(":#{match_by}", &__MODULE__.process_result/1)
+    {:reply, result, state}
   end
 
-  def handle_call({:sync_two_nodes, node_name1, node_name2}, _, state) do
-    port = state[node_name2].port
-    sync_port = state[node_name2].sync_port
-
-    cmd1 = "{:ok, peer_info} = Client.get_info(\"localhost:#{port}\")\n"
-    Port.command(state[node_name1].process_port, cmd1)
-
-    cmd2 = "pub_key = Map.get(peer_info, :peer_pubkey) |> Keys.peer_decode()\n"
-    Port.command(state[node_name1].process_port, cmd2)
-
-    cmd3 = "Peers.try_connect(%{host: 'localhost', port: #{sync_port}, pubkey: pub_key})\n"
-    Port.command(state[node_name1].process_port, cmd3)
-
+  def handle_call({:post, node, match_by, cmd}, _from, state) do
+    %{port_id: port_id} = Map.get(state, node)
+    command = "{:#{match_by}," <> cmd <> "}\n"
+    Port.command(port_id, command)
+    expected_result = fn _ -> :ok end
+    :ok = receive_result(":#{match_by}", expected_result)
     {:reply, :ok, state}
   end
 
-  def handle_call({:get_state}, _, state) do
-    {:reply, state, state}
-  end
-
-  def handle_call({:delete_all_nodes}, _, state) do
-    # killing all the processes and closing all of the ports of the nodes
-    Enum.each(state, fn {_, val} ->
-      Port.command(val.process_port, ":erlang.halt()\n")
-      Port.close(val.process_port)
+  def handle_call(:delete_nodes, _from, state) do
+    Enum.each(state, fn {_node, %{port_id: port_id, node_port: port}} ->
+      Port.command(port_id, ":erlang.halt()\n")
+      Port.close(port_id)
       path = String.replace(System.cwd(), ~r/(?<=elixir-node).*$/, "") <> "/apps/aecore/priv/"
-      File.rm_rf(path <> "aewallet_#{val.port}")
-      File.rm_rf(path <> "peerkeys_#{val.port}")
-      File.rm_rf(path <> "rox_db_#{val.port}")
+      File.rm_rf(path <> "aewallet_#{port}")
```

Comment:
we should move this to explicitly "test" named directories so this doesn't delete data used in development

Reply:
But when we are getting data for the keys, we are looking into these folders for already created keys. If we move it to somewhere else, it won't read from the correct place and possibly will create new keys in the /priv dir anyways

Reply:
but somewhere we set these directories to be used, so we can add some "test" string in the path